docker-images/kafka-based/kafka/scripts/kafka_readiness_kraft.sh

```diff
@@ -3,7 +3,10 @@ set -e
 
 file=/tmp/strimzi.properties
 test -f $file
-roles=$(grep -Po '(?<=^process.roles=).+' "$file")
+
+# During migration, the process.roles field can be still not set on broker only nodes
+# so, because grep would fail, the "|| true" operation allows to return empty roles result
+roles=$(grep -Po '(?<=^process.roles=).+' "$file" || true)
```

Comment:
Why do you expect to use the KRaft readiness if the roles are not present?

Reply:
During the `Migration` status, the broker nodes have both KRaft enabled and ZooKeeper (together with migration flag) but the `process.roles` is not set (it will be in the post migration). But because KRaft is already there, the KRaft readiness has to be used.